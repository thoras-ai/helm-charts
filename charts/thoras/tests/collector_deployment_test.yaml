suite: Collector
templates:
  - collector/deployment.yaml
tests:
  - it: Deployment name and images should be correct
    set:
      thorasVersion: TEST
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: metadata.name
          value: metrics-collector
      - equal:
          path: spec.template.spec.containers[0].image
          value: us-east4-docker.pkg.dev/thoras-registry/platform/elasticsearch:8.12.1
      - equal:
          path: spec.template.spec.containers[1].image
          value: us-east4-docker.pkg.dev/thoras-registry/platform/thoras-collector:TEST
  - it: Default containers should match snapshots
    set:
      thorasVersion: dev
    asserts:
      - matchSnapshot:
          path: spec.template.spec.containers[0]
      - matchSnapshot:
          path: spec.template.spec.containers[1]
  - it: Should mount pvc when in persistence mode
    set:
      metricsCollector:
        persistence:
          enabled: true
    asserts:
      - equal:
          path: spec.template.spec.volumes
          value:
            - name: elastic-search-data
              persistentVolumeClaim:
                claimName:  elastic-search-data
      - equal:
          path: spec.template.spec.containers[0].volumeMounts
          value:
            - mountPath: /usr/share/elasticsearch/data
              name: elastic-search-data
