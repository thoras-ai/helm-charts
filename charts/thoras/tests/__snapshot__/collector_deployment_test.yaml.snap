Default containers should match snapshots:
  1: |
    env:
      - name: ES_JAVA_OPTS
        value: -Xms750m -Xmx750m
      - name: discovery.type
        value: single-node
      - name: xpack.security.enabled
        value: "true"
      - name: bootstrap.memory_lock
        value: "false"
      - name: ELASTIC_PASSWORD
        valueFrom:
          secretKeyRef:
            key: password
            name: thoras-elastic-password
    image: us-east4-docker.pkg.dev/thoras-registry/platform/elasticsearch:8.12.1
    imagePullPolicy: IfNotPresent
    name: elasticsearch
    ports:
      - containerPort: 9200
  2: |
    args:
      - |
        # Don't start app until ES is available
        retries=0
        max_retries=48
        until [ $retries -eq $max_retries ]; do
          if curl -s --connect-timeout 5 --fail -XGET ${ES_HOST}; then
            echo "elasticsearch ready"
            break
          else
            echo "elasticsearch not ready"
            retries=$((retries+1))
            sleep 5
          fi
        done
        if [ $retries -eq $max_retries ]; then
            echo "fatal: giving up on elasticsearch availability"
            exit 1
        fi
        node index.js start --es=${ES_HOST}
    command:
      - /bin/sh
      - -c
    env:
      - name: THORAS_NS
        valueFrom:
          fieldRef:
            fieldPath: metadata.namespace
      - name: ES_HOST
        valueFrom:
          secretKeyRef:
            key: host
            name: thoras-elastic-password
      - name: SLACK_WEBHOOK_URL
        valueFrom:
          secretKeyRef:
            key: webhookUrl
            name: thoras-slack
      - name: SLACK_ERRORS_ENABLED
        value: "false"
      - name: LOGLEVEL
        value: info
    image: us-east4-docker.pkg.dev/thoras-registry/platform/thoras-collector:dev
    imagePullPolicy: IfNotPresent
    name: thoras-collector
